File: euler85.py
# No. This is not efficient.
# No. I don't care. 
# Yes. I am leaving it as it is. 
# The answer is right.

row = 1
col = 2
count = 0

# arbitrary end condition
while count < 1990000:
    # initialize values each iteration
    row += 1
    if row > 100:
        col += 1
        row = 2
    count = 0

    # determine the number of shapes per row/col based on initialized values
    # basically determines smaller "sub-grids" of the full grid that provides
    #  the number of the row x col shapes that fit
    for i in range(1, row + 1):
        r = 0
        for ii in range(1, row + 1):
            if ii + i <= row + 1:
                r += 1
        for j in range(1, col + 1):
            c = 0
            for jj in range(1, col + 1):
                if jj + j <= col + 1:
                    c += 1
            count += r*c

print(count)
print(row, col, row * col)

File: euler85.java
package ProjectEuler67;

public class euler85
{
    public static void main(String args[])
    {
        int row = 10;
        int col = 9;
        int count = 0;
    
        int target = 2000000;
        
        boolean flag = true;
    
        do
        {
            col++;
            if (col > 100)
            {
                col = 10;
                row++;
            }
            
            if (row > 100)
                flag = false;
            
            count = 0;
            
            for (int i = 1; i <= row; i++)
            {
                int r = 0;
                for (int ii = 1; ii <= row; ii++)
                    if (ii + i <= row + 1)
                        r++;
                for (int j = 1; j <= col; j++)
                {
                    int c = 0;
                    for (int jj = 1; jj <= col; jj++)
                        if (jj + j <= col + 1)
                            c++;
                    count = count + r*c;
                }
            }
            
            if (target - count == 2)
                flag = false;
            
        } while(flag);
    
        System.out.printf("count: %d\nrow: %d\ncol: %d\narea: %d\n", count, row, col, row*col);
    }
}
